django/db/models/sql/query.py

```diff
@@ -1137,7 +1137,29 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
 
         # Work out the lookup type and remove it from the end of 'parts',
         # if necessary.
-        value, lookups, used_joins = self.prepare_lookup_value(value, lookups, can_reuse, allow_joins)
+        if isinstance(value, (list, tuple)) and value:
+            klass = type(value)
+            new_value = []
+            new_lookups = []
+            new_used_joins = []
+            # Look up using the iterable of arguments first, in case it's an empty list
+            for inner_value in value:
+                inner_value, lookups, used_joins = self.prepare_lookup_value(inner_value,
+                    lookups,
+                    can_reuse,
+                    allow_joins)
+                new_value.append(inner_value)
+                for lookup in lookups:
+                    if lookup not in new_lookups:
+                        new_lookups.append(lookup)
+                for used_join in used_joins:
+                    if used_join not in new_used_joins:
+                        new_used_joins.append(used_join)
```

Comment:
Hi @akaariai,

I'm wondering if you could clarify this for me please? What about the aggregation of required joins from the independent from the sub-values would cause the join type to change? It seems to just be table names without metadata.

Matt